### PR TITLE
fix(datalist): actions wrapper

### LINKED ISSUE #10903: DataList: Actions spacing is off on non plain buttons
**Please describe the issue**
Actions spacing is off on non plain buttons.  

All actions should have the internal wrapper applied by the isPlainButtonAction
Prop.  we can then remove the prop.

**Are there visuals for this issue? Please provide screenshots**
Include screenshots or links to Marvel or other mockups.
https://staging-v6.patternfly.org/components/data-list#actions-single-and-multiple 

**Could you point to a branch or draft PR where this issue exists? Please provide a link to the code**

**Any other information?**

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #10903 

**Additional issues**:

### CODE REVIEW

File: packages/react-core/src/components/DataList/__tests__/DataListAction.test.tsx [hunk 1]
```diff
@@ -58,18 +57,19 @@ test(`Renders with class ${styles.dataListAction} when isPlainButtonAction = tru
...
-    expect(screen.getByText('test')).toHaveClass(styles.modifiers[`${visMod}OnLg`]);
-    expect(screen.getByText('test')).toHaveClass(styles.modifiers[`${visMod}OnXl`]);
-    expect(screen.getByText('test')).toHaveClass(styles.modifiers[`${visMod}On_2xl`]);
+    expect(screen.getByTestId('actions')).toHaveClass(styles.modifiers[`${visMod}OnSm`]);
+    expect(screen.getByTestId('actions')).toHaveClass(styles.modifiers[`${visMod}OnMd`]);
+    expect(screen.getByTestId('actions')).toHaveClass(styles.modifiers[`${visMod}OnLg`]);
+    expect(screen.getByTestId('actions')).toHaveClass(styles.modifiers[`${visMod}OnXl`]);
+    expect(screen.getByTestId('actions')).toHaveClass(styles.modifiers[`${visMod}On_2xl`]);
   });
 });
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/DataList/__tests__/DataListAction.test.tsx:75]: Can you add a test which asserts that the children of the DataListAction now have the dataListAction class?

**Comment by mcoker** [comment]
@andrew-ronaldson @lboehling just want to run this change by you and make sure this design change is OK. I created a codepen showing a regular data-list and one with the proposed change - https://codepen.io/mcoker/pen/PorgJab

Just as a reminder, the TL;DR of this change is to update all of the "action" cells in a data-list to have a negative top/bottom margin that is equal to a button's top/bottom padding. We already do this for plain actions (kebabs in the codepen), and this will do the same for non-plain actions. What that effectively does is keeps the presence of a button in a row from increasing the height of the row, and it aligns the button text with the data-list text.

On a row with a single line of text, this looks great IMO. It vertically centers the button in the row.

Before
Screenshot 2024-09-12 at 8 09 59 PM

After
Screenshot 2024-09-12 at 8 09 50 PM

But when text wraps and the row is taller, it's more obvious that non-plain actions are shifted up. I just want to confirm that's ok from y'alls perspective:

Before
Screenshot 2024-09-12 at 8 14 43 PM

After
Screenshot 2024-09-12 at 8 14 49 PM

**Review by mcoker — COMMENTED** [review]
Discussed offline but looks like we may need to handle this a little differently. I think we'll probably just remove `<div className={css(styles.dataListAction)}>` and update core to apply the margin change to `<div className={css(styles.dataListItemAction` instead. So in this PR, we'd just remove that first div and the CSS will handle the rest.

But first I want to [confirm with design](https://github.com/patternfly/patternfly-react/pull/10939#issuecomment-2347830104) that the proposed change is still 👍👍

**Comment by lboehling** [comment]
@mcoker i'm good w this change!

File: packages/react-core/src/components/DataList/DataListAction.tsx [hunk 2]
```diff
@@ -35,12 +33,10 @@ export const DataListAction: React.FunctionComponent<DataListActionProps> = ({
   id,
   'aria-label': ariaLabel,
   'aria-labelledby': ariaLabelledBy,
-
-  isPlainButtonAction,
   ...props
 }: DataListActionProps) => (
   <div className={css(styles.dataListItemAction, formatBreakpointMods(visibility, styles), className)} {...props}>
-    {isPlainButtonAction ? <div className={css(styles.dataListAction)}>{children}</div> : children}
+    <div className={css(styles.dataListAction)}>{children}</div>
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DataList/DataListAction.tsx:39]: This should be good to update now that https://github.com/patternfly/patternfly/pull/7089 has been merged in core

**Review by kmcfaul — APPROVED** [review]
lgtm

**Review by thatblindgeye — APPROVED** [review]

**Review by mcoker — APPROVED** [review]
LPTM! 🎸🤘

**Pull request merged**